GCC dangling reference false positive

GCC 14 currently doesn't compile sanicdns because of a new dangling reference detection feature that produces a lot of false positives. Example compiler error:

```FAILED: CMakeFiles/sanicdns_libs.dir/src/net_info.cpp.o
/usr/bin/g++ -DNIC_AF_XDP -DSPDLOG_COMPILED_LIB -DSPDLOG_NO_EXCEPTIONS -I/home/geert/projects/sanicdns/include -I/home/geert/projects/sanicdns/utils -I/home/geert/projects/sanicdns/lib/spdlog/include -isystem /usr/include/dbus-1.0 -isystem /usr/lib/x86_64-linux-gnu/dbus-1.0/include -isystem /home/geert/projects/sanicdns/lib/glaze/include -Wall -Werror -mssse3 -msse4.2 -g -fsanitize=address -g -std=gnu++20 -Wno-volatile -Wno-missing-braces -MD -MT CMakeFiles/sanicdns_libs.dir/src/net_info.cpp.o -MF CMakeFiles/sanicdns_libs.dir/src/net_info.cpp.o.d -o CMakeFiles/sanicdns_libs.dir/src/net_info.cpp.o -c /home/geert/projects/sanicdns/src/net_info.cpp
In file included from /home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/fmt.h:24,
                 from /home/geert/projects/sanicdns/lib/spdlog/include/spdlog/common.h:45,
                 from /home/geert/projects/sanicdns/lib/spdlog/include/spdlog/spdlog.h:12,
                 from /home/geert/projects/sanicdns/src/net_info.cpp:9:
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h: In instantiation of ‘constexpr fmt::v8::detail::value<Context> fmt::v8::detail::make_arg(T&&) [with bool IS_PACKED = true; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; type <anonymous> = fmt::v8::detail::type::string_type; T = FixedName<16>&; typename std::enable_if<IS_PACKED, int>::type <anonymous> = 0]’:
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h:1855:77:   required from ‘constexpr fmt::v8::format_arg_store<Context, Args>::format_arg_store(T&& ...) [with T = {FixedName<16>&}; Context = fmt::v8::basic_format_context<fmt::v8::appender, char>; Args = {FixedName<16>}]’
 1853 |         data_{detail::make_arg<
      |               ~~~~~~~~~~~~~~~~~
 1854 |             is_packed, Context,
      |             ~~~~~~~~~~~~~~~~~~~
 1855 |             detail::mapped_type_constant<remove_cvref_t<T>, Context>::value>(
      |             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
 1856 |             std::forward<T>(args))...} {
      |             ~~~~~~~~~~~~~~~~~~~~~~
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h:1872:38:   required from ‘constexpr fmt::v8::format_arg_store<Context, typename std::remove_cv<typename std::remove_reference<Args>::type>::type ...> fmt::v8::make_format_args(Args&& ...) [with Context = basic_format_context<appender, char>; Args = {FixedName<16>&}]’
 1872 |   return {std::forward<Args>(args)...};
      |                                      ^
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h:3119:44:   required from ‘std::string fmt::v8::format(format_string<T ...>, T&& ...) [with T = {FixedName<16>&}; std::string = std::__cxx11::basic_string<char>; format_string<T ...> = basic_format_string<char, FixedName<16>&>]’
 3119 |   return vformat(fmt, fmt::make_format_args(args...));
      |                       ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
/home/geert/projects/sanicdns/src/net_info.cpp:217:36:   required from here
  217 |                 return tl::unexpected(fmt::format("Cannot get index for {}", iface));
      |                                       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h:1706:15: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));
      |               ^~~
/home/geert/projects/sanicdns/lib/spdlog/include/spdlog/fmt/bundled/core.h:1706:46: note: the temporary was destroyed at the end of the full expression ‘fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >().fmt::v8::detail::arg_mapper<fmt::v8::basic_format_context<fmt::v8::appender, char> >::map<FixedName<16> >((*(const FixedName<16>*)(& std::forward<FixedName<16>&>((* & val)))))’
 1706 |   const auto& arg = arg_mapper<Context>().map(std::forward<T>(val));```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,6 +69,10 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 # Find DPDK
 find_package(dpdk REQUIRED)
 
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER "14.0.0")
+	    add_compile_options(-Wno-dangling-reference)
+endif()
+
 # Force Colored Output
 if (${FORCE_COLORED_OUTPUT})
     if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")

